File: iosApp/iosApp/FeedView.swift
import SwiftUI
import shared

struct FeedView: View {
    let service = FeedService()

    @State var items = [FeedItem]()

    var body: some View {
        ScrollView {
            LazyVStack {
                ForEach(items, id: \.id, content: makeItemView)
            }
        }
        .task {
            items = try! await service.loadItems()
        }
    }

    func makeItemView(for item: FeedItem) -> some View {
        VStack(alignment: .leading) {
            GeometryReader { geometry in
                AsyncImage(url: item.image.flatMap { URL(string: $0.url) })
                    .frame(width: geometry.size.width, height: geometry.size.height)
                    .clipped()
            }
            .frame(height: 100)

            Text(item.title).font(.title)
            Text(item.author).font(.caption)
            Text(item.description_).font(.body)
                .padding(.top, 4)
        }
        .padding()
    }
}

struct FeedView_Previews: PreviewProvider {
    static var previews: some View {
        FeedView()
    }
}

File: consoleApp/src/main/kotlin/cz/root/reader/console/FeedView.kt
package cz.root.reader.console

import androidx.compose.runtime.Composable
import com.jakewharton.mosaic.Column
import com.jakewharton.mosaic.Text
import cz.root.reader.FeedItem

@Composable
fun FeedView(feedItems: List<FeedItem>) {
    Column {
        feedItems.forEach {
            FeedItemView(it)
        }
    }
}

@Composable
private fun FeedItemView(item: FeedItem) {
    Text(value = item.title)
    Text(value = item.author)
    Text(value = item.description)
    Text(value = "-----------------------")
}